### DISCUSSION TITLE
CSS-in-JS solution that supports SWC and PostCSS?

### DISCUSSION THREAD

**Discussion by mattjennings** [vercel/next.js#35160] [Help]
Are there any CSS-in-JS solutions that work with SWC and support PostCSS? As far as I know, the only CSS-in-JS libraries that work with SWC are styled-jsx and styled-components. 

styled-jsx supported PostCSS via babel plugins, but [I don't know if there's a plan to support it in SWC](https://github.com/vercel/styled-jsx/issues/782). I don't think styled-components ever had official support for PostCSS, so I don't have much hope there.

I'm curious if there are any other options, or if there are plans to add PostCSS support to either library. I know I can use CSS modules, but I'd _really_ like a CSS-in-JS option.

**Comment by tettoffensive** [comment]
I'm kinda wondering this myself. My current CSS-in-JS solution is using `emotion` as well as [twin.macro](https://github.com/ben-rogerson/twin.macro) for `tailwind` support. I was kinda feeling like since `Next.js` already comes with `styled-jsx` dropping the additional dependency of `emotion`. But then I'd have to rework my usage of `tailwind` with `twin.macro` and I often use `scss` syntax. It doesn't seem like `styled-jsx` will support SWC with plugins at the moment.

So it seems like I'm stuck with `babel` for now.
